src/unit_tests/ctx/delete_account.rs

```diff
@@ -0,0 +1,158 @@
+                email: "user_email".to_owned(),
+                fb_id: None,
+                avatar: None,
+                last_modified: TestEnv::now(),
+                date_registered: TestEnv::now(),
+                trakt: None,
+                premium_expire: None,
+                gdpr_consent: GDPRConsent {
+                    tos: true,
+                    privacy: true,
+                    marketing: true,
+                    from: Some("tests".to_owned()),
+                },
+            },
+        }),
+        ..Default::default()
+    };
+
+    let _env_mutex = TestEnv::reset().expect("Should have exclusive lock to TestEnv");
+    *FETCH_HANDLER.write().unwrap() = Box::new(fetch_handler);
+
+    STORAGE.write().unwrap().insert(
+        PROFILE_STORAGE_KEY.to_owned(),
+        serde_json::to_string(&profile).unwrap(),
+    );
+
+    let (runtime, _rx) = Runtime::<TestEnv, _>::new(
+        TestModel {
+            ctx: Ctx::new(
+                profile,
+                LibraryBucket::default(),
+                StreamsBucket::default(),
+                ServerUrlsBucket::new::<TestEnv>(None),
+                NotificationsBucket::new::<TestEnv>(None, vec![]),
+                SearchHistoryBucket::default(),
+                DismissedEventsBucket::default(),
+            ),
+        },
+        vec![],
+        1000,
+    );
+
+    TestEnv::run(|| {
+        runtime.dispatch(RuntimeAction {
+            field: None,
+            action: Action::Ctx(ActionCtx::DeleteAccount("password".to_owned())),
+        })
+    });
+
+    assert_eq!(
+        runtime.model().unwrap().ctx.profile,
+        Default::default(),
+        "profile updated successfully in memory"
+    );
+
+    assert!(
+        STORAGE
+            .read()
+            .unwrap()
+            .get(PROFILE_STORAGE_KEY)
+            .map_or(false, |data| {
+                serde_json::from_str::<Profile>(data).unwrap() == Default::default()
+            }),
+        "profile updated successfully in storage"
+    );
+
+    assert_eq!(
+        REQUESTS.read().unwrap().len(),
+        2,
+        "Two requests have been sent"
+    );
+
+    assert_eq!(
+        REQUESTS.read().unwrap().get(0).unwrap().to_owned(),
+        Request {
+            url: "https://api.strem.io/api/deleteUser".to_owned(),
+            method: "POST".to_owned(),
+            body: "{\"type\":\"DeleteAccount\",\"authKey\":\"auth_key\",\"password\":\"password\"}"
+                .to_owned(),
+            ..Default::default()
+        },
+        "Delete account request has been sent"
+    );
+
+    assert_eq!(
+        REQUESTS.read().unwrap().get(1).unwrap().to_owned(),
+        Request {
+            url: "https://api.strem.io/api/logout".to_owned(),
+            method: "POST".to_owned(),
+            body: "{\"type\":\"Logout\",\"authKey\":\"auth_key\"}".to_owned(),
+            ..Default::default()
+        },
+        "Logout request has been sent"
+    );
```

Comment:
Just to double check:
The API will still be able to Log us out using the same auth key if we first delete the account?

Reply:
This is a downside from this: https://github.com/Stremio/stremio-core/pull/753/files#diff-5e5b6a800019c0be2c32e04c71c27cda593eaef5d4b50e2a6772eb6e8cffac88R397
It will dispatch the `Internal:Logout` message so that every update functions can remove the user data
But this message is also handled in the ctx model which trigger a delete session / logout request
https://github.com/Stremio/stremio-core/blob/354f6b1734e37a44ea23ed99234d22b90daa7bfa/src/models/ctx/ctx.rs#L119